storage/store.go

```diff
@@ -344,6 +322,7 @@ func NewStore(ctx StoreContext, eng engine.Engine) *Store {
 		engine:      eng,
 		allocator:   newAllocator(sf.findStores),
 		ranges:      map[int64]*Range{},
+		nodeDesc:    nodeDesc,
```

Comment:
We should add a check somewhere that Store.nodeDesc.NodeID is consistent with Store.Ident.NodeID. This would have made the multiTestContext problem easier to spot. I think Store.Start would be the right place for this.